app/invocation/invocation_action_tree_node.tsx

```diff
@@ -28,11 +32,12 @@ function getChildCountText(childCount: Number) {
   return format.formatWithCommas(childCount) + " children";
 }
 
-export default class InputNodeComponent extends React.Component<Props, State> {
-  render() {
-    const digestString = this.props.node.obj.digest?.hash ?? "";
+export default class TreeNodeComponent extends React.Component<Props, State> {
+  renderDigestNode(node: FileNode | DirectoryNode) {
```

Comment:
nit: maybe call this `renderFileOrDirectoryNode`

Reply:
Done. Great suggestion!